R/s4.R

```diff
@@ -0,0 +1,59 @@
+#' Checks if the passed entity is an S4 instance of a specified class.
+#'
+#' @param value the value to check
+#' @param class_name the name of the class, as a string.
+#' @param allow_null if true, accept a value that is NULL.
+#'
+#' @examples
+#' \dontrun{
+#' # For assertion
+#' assertthat::assert_that(
+#'   qscheck::is_s4_instance(value, "MyClass")
+#'   )
+#' # For check
+#' if (qscheck::is_s4_instance(value, "MyClass")) {}
+#' }
+#'
+#' @concept oop
+#' @export
+is_s4_instance <- function(value, class_name, allow_null = FALSE) {
+
+  if (is.null(value) && allow_null) {
+    return(TRUE)
```

Comment:
I need to fix this... it's kind of ugly. The problem is that this returns TRUE only if it's both NULL _and_ the allow_null is TRUE. if allow_null is false, then it will carry on, eventually stumbling on other checks later on, but it would be better to just return FALSE immediately.

In other words, this need to do if (is.null(value) { return(allow_null) }

The same "error" is present elsewhere. Leave it for now, but create an issue about it so I'll tackle all of them at the same time.